### ISSUE TITLE
ConcateningMedia, RepeatOne and next / previous

### ISSUE THREAD

**Issue by tomislavturcic** [google/ExoPlayer#5602]
I have a question regarding REPEAT_ONE mode, ConcateningMedia and using SimpleExoPlayer.next(), previous() and related methods.

From my understanding, REPEAT_ONE should function by endlessly looping same MediaSource. From what I see in the source code of Timeline.getNextWindowIndex, it should return currentWindowIndex when repeat mode is REPEAT_ONE. But this isn't the case. Instead using simpleExoPlayer.next() moves to next window index and plays next MediaSource. 

In short, I have a ConcateningMedia with MediaSourceA, MediaSourceB and MediaSourceC.
When I set REPEAT_ONE and play MediaSourceA, I expect getPreviousWindowIndex() and getNextWindowIndex() to return same values as getCurrentWindowIndex() and that calling next() replays MediaSourceA. Instead, MediaSourceB is played.
 
Is this expected behavior?

**Comment by tonihei** [comment]
Yes, that is expected behaviour. 

The idea is that you can have UI elements which let you skip to the next and previous item (using `player.next()` and `player.previous()` even though you have REPEAT_ONE enabled. The current item is still repeated endlessly without any interaction. The `getNextWindowIndex()` and `getPreviousWindowIndex()` methods return matching values to tell you which window index will be played when calling `next()` or `previous()`.

**Comment by tomislavturcic** [comment]
Ok, thanks for clarification!

**Comment by tonihei** [comment]
If you'd like to know the next item the player will play automatically, you can use `player.getCurrentTimeline().getNextWindowIndex(windowIndex, repeatMode, shuffleMode)` to query any transition under different repeat and shuffle mode conditions.

**Comment by ojw28** [comment]
`Player` Javadoc will be improved to more explicitly explain this behavior in a future release. Removing the `documentation candidate` label as these changes have been made internally.
